solution/backend/content_search/models.py

```diff
@@ -0,0 +1,138 @@
+    pass
+
+
+class ContentIndex(models.Model):
+    doc_name_string = models.CharField(max_length=512, null=True, blank=True)
+    summary_string = models.TextField(null=True, blank=True)
+    file_name_string = models.CharField(max_length=512, null=True, blank=True)
+    date_string = VariableDateField()
+    content = models.TextField()
+    url = models.CharField(max_length=512, blank=True, null=True)
+    subjects = models.ManyToManyField(Subject, blank=True, related_name="content")
+    document_type = models.ForeignKey(DocumentType, blank=True, null=True, related_name="content", on_delete=models.SET_NULL)
+    category = models.ForeignKey(
+        AbstractCategory, null=True, blank=True, on_delete=models.SET_NULL, related_name="content"
+    )
+    locations = models.ManyToManyField(AbstractLocation, blank=True, related_name="content", verbose_name="Regulation Locations")
+    resource_type = models.CharField(max_length=25, null=True, blank=True)
+    file = models.ForeignKey(UploadedFile, blank=True, null=True, on_delete=models.CASCADE)
+    supplemental_content = models.ForeignKey(SupplementalContent, blank=True, null=True, on_delete=models.CASCADE)
+    fr_doc = models.ForeignKey(FederalRegisterDocument, blank=True, null=True, on_delete=models.CASCADE)
+    objects = ContentIndexManager()
+
+
+def create_search(updated_doc, file=None):
+    file_content = ''
+    if file:
+        file_content = file.content
+    else:
+        # Trigger lambda here to get the text
+        file_content = ''
+    fi = establish_content_type(updated_doc)
+
+    fi.content = file_content
+    fi.save()
+    fi.locations.set(updated_doc.locations.all())
+    fi.subjects.set(updated_doc.subjects.all())
+    fi.save()
+
+
+def establish_content_type(updated_doc):
+    if isinstance(updated_doc, UploadedFile):
+        return ContentIndex(
+            file=updated_doc,
+            document_type=updated_doc.document_type,
+            file_name_string=updated_doc.file_name,
+            url=updated_doc.uid,
+            doc_name_string=updated_doc.document_name,
+            summary_string=updated_doc.summary,
+            date_string=updated_doc.date,
+            resource_type='internal'
+        )
+    if isinstance(updated_doc, SupplementalContent) or isinstance(updated_doc, FederalRegisterDocument):
+        ci = ContentIndex(
+            category=updated_doc.category,
+            url=updated_doc.url,
+            doc_name_string=updated_doc.name,
+            summary_string=updated_doc.description,
+            date_string=updated_doc.date,
+            resource_type='external'
+        )
+
+        if isinstance(updated_doc, SupplementalContent):
+            ci.supplemental_content = updated_doc
+        else:
+            ci.fr_doc = updated_doc
+        return ci
+    return None
+
+
+def update_search(sender, instance, created, **kwargs):
+    try:
+        if isinstance(instance, UploadedFile):
```

Comment:
we are trying to find the index related to the object here.  If it cant find it, it will throw an exception and then will create it